### PR TITLE
fix: Refactor faster page loads

### LINKED ISSUE #25586: [Task]: First page load optimisation refactor
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Refactor task to improve the routing architecture by shifting page load data fetching responsibility from the main `App Router` to the individual responsible routes.

This refactor is to be exercised on top of the changes introduced by this [PR](https://github.com/appsmithorg/appsmith/pull/24365)

[Detailed explanation](https://www.notion.so/appsmith/Page-load-API-optimisation-14aad3c189424ab984922276bc3d27bc)

### PR DESCRIPTION
**Pull request by hetunandu**
## Description
Improve page loads by removing the blocking calls of user and tenant information. These calls were called before any other routes would load and that would increase the time taken for other subsequent calls.

Reasons it was needed for 
1.  Branding info comes from tenant api call. This being executed later would cause flashing of colours if branding is updated
2.  Licence info comes from tenant api call. A check is needed for licence validity and show appropriate screens instead of the app for users
3. User info was needed to show the correct admin settings to the user

We are going to mitigate dependency 1 and 2 by hiding the ui by making the opacity to 0 till the responses are not received. This will ensure users don't see the screen, yet the JS is being loaded efficiently and api calls are done earlier.
We will move the admin settings dependency to the route itself so that other routes do not delay because of this change

#### PR fixes following issue(s)
Fixes #25586

#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
Needs regression testing on EE with admin settings, branding and licence checks
Page loads on CE and EE

- [x] Manual
- [ ] Cypress

#### Test Plan
EE:
- [ ] admin settings
- [ ] branding
- [ ] licence checks
- [ ] Page loads

CE
- [x] admin settings
- [x] Page loads
>
#### Issues raised during DP testing
No issues found during DP (CE) Testing

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/public/index.html [hunk 1]
```diff
@@ -91,8 +94,10 @@
     To keep zIndex for tooltips higher than app comments, todo remove when migrating to Tooltip2
     Currently the className does not apply to the portal root, so we're unable to work with z-indexes based on that
   -->
-  <div id="header-root"></div>
-  <div id="root"></div>
+  <div id="app-load-bock">
```
Comment (dipyamanbiswas07) [inline app/client/public/index.html:97]: should this be app-load-block?

Comment (hetunandu) [reply app/client/public/index.html]: Fixed

**Review by ashit-rath — APPROVED** [review]

File: app/client/src/pages/loaderHooks.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      const loader = document.getElementById("loader") as HTMLDivElement;
+      const appLoadBlock = document.getElementById(
+        "app-load-block",
+      ) as HTMLDivElement;
+
+      if (loader) {
+        loader.style.width = "100vw";
+
+        setTimeout(() => {
+          loader.style.opacity = "0";
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/loaderHooks.ts:40]: should we use opacity or something like visibility:hidden?

Comment (hetunandu) [reply app/client/src/pages/loaderHooks.ts:40]: Just found out opacity keeps the elements "interact-able" and listen to events. Visibility is a better option

**Review by dipyamanbiswas07 — DISMISSED** [review]
LGTM. Please create a DP on EE as well and test the license flows too.

**Comment by kavitasmoolya** [comment]
@hetunandu 
Tested page load and admin settings on [DP]( https://ce-25983.dp.appsmith.com/) pointing to CE and Looks good to me.
cc @ramsaptami

File: app/client/cypress.config.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { defineConfig } from "cypress";
 
 export default defineConfig({
+  projectId: "eyxvp8",
```
Comment (jsartisan) [inline app/client/cypress.config.ts:4]: shouldn't this come from GitHub secrets?

**Review by jsartisan — APPROVED** [review]

File: app/client/cypress.config.ts (hunk 3 above)
Comment (hetunandu) [reply app/client/cypress.config.ts:4]: It is a our public cypress project id. But it does not belong here. I will remove it

**Pull request merged**